### DISCUSSION TITLE
Accessing camera image from flowdeck and disabling auto height adjustment while using flow deck

### DISCUSSION THREAD

**Discussion by MAHASHANA** [bitcraze/discussions#1488] [Q&A]
Hi Team,
I would like to know if there is a way to access camera data from flow deck 2.0. And Is there a way to stop the automatic height adjustment for the drone when the system detects a change in surface height. I came across this https://www.bitcraze.io/2022/01/a-modified-ekf-for-constant-altitude-flights/ but just wanted to confirm if its the only way?

**Comment by knmcguire** [comment]
Currently there is no API to grab the image of the PMW3901 but you can perhaps browse the internet if people have managed to do so. It will probably be very slow and no optical flow can be detected during that grab

And also there is no detection for height adjustment, but you could check the code of the UKF in the firmware as that has a nice outlier rejection filter, but it will not change the height I'm afraid.
